src/traced/probes/sys_stats/sys_stats_data_source.h

```diff
@@ -103,9 +103,12 @@ class SysStatsDataSource : public ProbesDataSource {
   void ReadThermalZones(protos::pbzero::SysStats* sys_stats);
   void ReadCpuIdleStates(protos::pbzero::SysStats* sys_stats);
   void ReadGpuFrequency(protos::pbzero::SysStats* sys_stats);
-  std::optional<uint64_t> ReadAMDGpuFreq();
+  std::optional<uint64_t> ReadAmdGpuFreq();
+
+  void OpenGpuFreqFiles(OpenFunction open_fn);
 
   size_t ReadFile(base::ScopedFile*, const char* path);
+  std::optional<uint64_t> ReadFileToUInt64(base::ScopedFile*, const char* path);
```

Comment:
can you expand a bit on what's the rationale here?
we have a virtual method above ReadFileToUInt64(const std::string& path) but you are adding a non-virtual here. who is supposed to use what? when?

I think this is a bit confusing as I can see people using one or the other, and perhaps messing up tests. what's the logic here?

Also if this is really a private impl detail it should go into the anonymous namespace in the .cc file instead?